File: backend/app/schemas/project.py
"""项目相关模式"""
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from decimal import Decimal


class ProjectBase(BaseModel):
    """项目基础模式"""
    name: str = Field(..., description="项目名称", max_length=100)
    description: Optional[str] = Field(None, description="项目描述")
    estimated_output_value: Optional[Decimal] = Field(None, description="预计产值（元）")


class ProjectCreate(ProjectBase):
    """创建项目模式"""
    pass


class ProjectUpdate(BaseModel):
    """更新项目模式"""
    name: Optional[str] = Field(None, description="项目名称", max_length=100)
    description: Optional[str] = Field(None, description="项目描述")
    estimated_output_value: Optional[Decimal] = Field(None, description="预计产值（元）")


class ProjectManagersUpdate(BaseModel):
    """协办项目经理列表（不包含创建者；仅创建者可维护）"""

    user_ids: List[int] = Field(default_factory=list, description="用户ID列表，仅能包含具备项目经理角色的用户")


class ProjectResponse(ProjectBase):
    """项目响应模式"""
    id: int
    created_by: int
    created_at: datetime
    updated_at: datetime
    creator_name: Optional[str] = None  # 创建者名称
    manager_user_ids: List[int] = Field(default_factory=list, description="协办项目经理用户ID列表")

    class Config:
        from_attributes = True


class ProjectListResponse(BaseModel):
    """项目列表响应模式"""
    total: int
    items: list[ProjectResponse]

File: frontend/src/api/project.ts
/**
 * 项目API
 */
import request from './index'

export interface Project {
  id: number
  name: string
  description?: string
  estimated_output_value?: number
  created_by: number
  creator_name?: string
  created_at: string
  updated_at: string
}

export interface ProjectCreate {
  name: string
  description?: string
  estimated_output_value?: number
}

export interface ProjectUpdate {
  name?: string
  description?: string
  estimated_output_value?: number
}

export interface ProjectListResponse {
  total: number
  items: Project[]
}

/**
 * 获取项目列表
 */
export function getProjects(params?: {
  creator_id?: number
  skip?: number
  limit?: number
}): Promise<ProjectListResponse> {
  return request.get('/api/v1/projects', { params })
}

/**
 * 获取项目详情
 */
export function getProject(projectId: number): Promise<Project> {
  return request.get(`/api/v1/projects/${projectId}`)
}

/**
 * 创建项目
 */
export function createProject(data: ProjectCreate): Promise<Project> {
  return request.post('/api/v1/projects', data)
}

/**
 * 更新项目
 */
export function updateProject(projectId: number, data: ProjectUpdate): Promise<Project> {
  return request.put(`/api/v1/projects/${projectId}`, data)
}

/**
 * 删除项目
 */
export function deleteProject(projectId: number): Promise<void> {
  return request.delete(`/api/v1/projects/${projectId}`)
}

export interface ProjectTask {
  id: number
  title: string
  description?: string
  status: string
  creator_id: number
  creator_name?: string
  assignee_id?: number
  assignee_name?: string
  estimated_man_days: number
  actual_man_days?: number
  required_skills?: string
  deadline?: string
  is_pinned: boolean
  created_at?: string
  updated_at?: string
  schedule?: {
    start_date?: string
    end_date?: string
    work_days: number
    is_pinned: boolean
  }
}

export interface ProjectTaskExecutionResponse {
  project_id: number
  project_name: string
  tasks: ProjectTask[]
  status_summary: Record<string, number>
  total: number
}

/**
 * 获取项目任务执行视图数据
 */
export function getProjectTasks(
  projectId: number,
  params?: {
    status?: string
    assignee_id?: number
    keyword?: string
  }
): Promise<ProjectTaskExecutionResponse> {
  return request.get(`/api/v1/projects/${projectId}/tasks`, { params })
}
